src/components/Dialogs/BoardDialog.vue

```diff
@@ -101,22 +103,24 @@ const formData = ref<{ name: string; columns: string[] }>({
 })
 const formNameError = ref(false)
 
-const submit = async () => {
-  if (formData.value.name === '') return
-
-  emits('close-dialog')
-
-  if (props.action === 'add') {
-    await addNewBoard({
+const submitFunctions = {
+  add: () =>
+    addNewBoard({
+      userID: userID.value,
+      name: formData.value.name,
+      columns: formData.value.columns.map((name) => ({ name, tasks: [] }))
+    }),
+  edit: () =>
+    editBoard({
```

Comment:
I'm not a fan of this solution. Submit function shouldn't be an object property. Please split it into separate functions, it's cleaner this way and more easy to debug.

Reply:
Why shouldn't it be an object property? If your next sentence is the answer for that, my follow-up question is: in what way is it "cleaner and more easy to debug"?